grammar.js

```diff
@@ -566,8 +565,8 @@ module.exports = grammar({
     )),
 
     pointer_expression: $ => choice(
-      prec.left(PREC.UNARY, seq('*', $._expression)),
-      prec.left(PREC.UNARY, seq('&', $._expression))
+      prec.left(PREC.CAST, seq('*', $._expression)),
+      prec.left(PREC.CAST, seq('&', $._expression))
```

Comment:
Maybe we should introduce a new `PREC.POINTER` constant here, rather than using `CAST`? Or do you think it's good to explicitly state that pointer operations and casts have the same precedence?

Reply:
According to this doc:
https://en.cppreference.com/w/c/language/operator_precedence
cast and deref should have the same precedence.

Reply:
Oh, ok great, thanks for explaining. Let's leave it exactly as you have it.